jlm/tooling/CommandLine.cpp

```diff
@@ -879,21 +892,50 @@ JlmOptCommandLineParser::ParseCommandLineArguments(int argc, char ** argv)
               "Write theta-gamma inversion statistics to file.")),
       cl::desc("Write statistics"));
 
+#ifdef ENABLE_MLIR
+  auto llvmInputFormat = JlmOptCommandLineOptions::InputFormat::Llvm;
+  auto mlirInputFormat = JlmOptCommandLineOptions::InputFormat::Mlir;
+
+  cl::opt<JlmOptCommandLineOptions::InputFormat> inputFormat(
+      "input-format",
+      cl::desc("Select input format:"),
+      cl::values(
+          ::clEnumValN(
+              llvmInputFormat,
+              JlmOptCommandLineOptions::ToCommandLineArgument(llvmInputFormat),
+              "Input LLVM IR [default]"),
+          ::clEnumValN(
+              mlirInputFormat,
+              JlmOptCommandLineOptions::ToCommandLineArgument(mlirInputFormat),
+              "Input MLIR")),
+      cl::init(llvmInputFormat));
+#endif
+
   auto llvmOutputFormat = JlmOptCommandLineOptions::OutputFormat::Llvm;
   auto xmlOutputFormat = JlmOptCommandLineOptions::OutputFormat::Xml;
+#ifdef ENABLE_MLIR
+  auto mlirOutputFormat = JlmOptCommandLineOptions::OutputFormat::Mlir;
+#endif
```

Comment:
In this case, I would inline `mlirOutputFormat` to avoid the `#ifdef`.

Reply:
Inlined